assets/src/blocks/Counter/CounterFrontend.js

```diff
@@ -0,0 +1,123 @@
+import { Component, Fragment } from '@wordpress/element';
+
+export class CounterFrontend extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      remaining: 0,
+      completed: 0
+    };
+
+    this.calculateRemaining = this.calculateRemaining.bind(this);
+    this.getCounterText = this.getCounterText.bind(this);
+  }
+
+  componentDidMount() {
+    // Calculate completed and remaining values depending on props
+    this.calculateRemaining();
+  }
+
+  componentDidUpdate({ target: prevTarget, completed: prevCompleted, completed_api: prevCompletedApi }) {
+    // Update completed and remaining values depending on props
+    const { target, completed, completed_api } = this.props;
+    if (target !== prevTarget || completed !== prevCompleted || completed_api !== prevCompletedApi) {
+      this.calculateRemaining();
+    }
+  }
+
+  calculateRemaining() {
+    const { completed_api } = this.props;
+    const target = this.props.target || 0;
+    let completed = this.props.completed || 0;
+    let remaining = 0;
+    if (completed_api) {
+      fetch(completed_api)
+        .then(response => response.json())
+        .then(({ unique_count }) => {
+          if (unique_count) {
+            completed = unique_count;
+            this.setState({
+              completed,
+              remaining: target - completed
```

Comment:
I can set a completed number higher that the target, and I guess an API could give that too, but in that case I think `remaining` shouldn't go negative, it should just stay at 0. So I would do `Math.max(target - completed, 0)` (or `completed >= target ? target - completed : 0`), and same line 47

Reply:
Good point, it was actually the existing behaviour! I'll fix it 👍

Reply:
I'm actually seeing just now that it's also possible to add negative values for `target` and `completed` 🙈 gonna fix that too